services/helper/mentees.js

```diff
@@ -1,34 +1,251 @@
             }
-        })
+
+            return common.successResponse({ statusCode: httpStatusCode.ok, message: apiResponses.SESSION_FETCHED_SUCCESSFULLY, result: sessions });
+        } catch (error) {
+            throw error;
+        }
     }
 
-    static reports(userId) {
-        return new Promise(async (resolve,reject) => {
+    static async reports(userId, filterType) {
+        let filterStartDate;
+        let filterEndDate;
+        let totalSessionEnrolled;
+        let totalsessionsAttended;
+        let filters;
+        try {
+            if (filterType === 'MONTHLY') {
+                [filterStartDate, filterEndDate] = utils.getCurrentMonthRange();
+            } else if (filterType === 'WEEKLY') {
+                [filterStartDate, filterEndDate] = utils.getCurrentWeekRange();
+            } else if (filterType === 'QUARTERLY') {
+                [filterStartDate, filterEndDate] = utils.getCurrentQuarterRange();
+            }
+
+            /* totalSessionEnrolled */
+            filters = {
+                createdAt: {
+                    $gte: filterStartDate.toISOString(),
+                    $lte: filterEndDate.toISOString()
+                },
+                userId: ObjectId(userId),
+                deleted: false
+            };
+
+            totalSessionEnrolled = await sessionAttendees.countSessionAttendees(filters);
+
+            /* totalSessionAttended */
+            filters = {
+                'sessionDetail.startDateUtc': {
+                    $gte: filterStartDate.toISOString(),
+                    $lte: filterEndDate.toISOString()
+                },
+                userId: ObjectId(userId),
+                deleted: false,
+                isSessionAttended: true
+            };
+
+            totalsessionsAttended = await sessionAttendees.countSessionAttendeesThroughStartDate(filters);
+            return common.successResponse({ statusCode: httpStatusCode.ok, message: apiResponses.MENTEES_REPORT_FETCHED_SUCCESSFULLY, result: { totalSessionEnrolled, totalsessionsAttended } });
+
+        } catch (error) {
+            throw error;
+        }
+    }
+
+    static async homeFeed(userId, isAMentor) {
+        try {
+            /* All Sessions */
+            const page = 1;
+            let limit = 4;
+            let allSessions = await this.sessions(userId, false, page, limit);
+
+            /* My Sessions */
+            limit = 2;
+            let mySessions = await this.sessions(userId, true, page, limit);
+
+            const result = {
+                allSessions: allSessions.result[0].data,
+                mySessions: mySessions.result[0].data,
+            }
+
+            const feedbackData = await feedbackHelper.pending(userId, isAMentor)
+
+            return common.successResponse({
+                statusCode: httpStatusCode.ok,
+                message: apiResponses.SESSION_FETCHED_SUCCESSFULLY,
+                result: result,
+                meta: {
```

Comment:
@rakeshSgr Let's put meta inside result.

Reply:
@aman-tunerlabs As meta information is not related to result of the API.
Currently we are passing feedback pending information, future  it may be configs or forms to support to front end app. 
Instead of calling the saparate API.